### PR TITLE
#216. Add full screen mode to WebView.

### PR DESCRIPTION
**Pull request by bogdan-evtushenko**

### CODE REVIEW

**Review by yevhenii-kanivets — APPROVED** [review]
Thanks for the fix 🙏

File: app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ * @param activityNonVideoView A View in the activity's layout that contains every other view that should be hidden when the video goes full-screen.
+ * @param activityVideoView A ViewGroup in the activity's layout that will display the video. Typically you would like this to fill the whole layout.
+ * @param loadingView A View to be shown while the video is loading (typically only used in API level <11). Must be already inflated and without a parent view.
+ * @param webView The owner VideoEnabledWebView. Passing it will enable the com.bogdan.codeforceswatcher.components.VideoEnabledWebChromeClient to detect the HTML5 video ended event and exit full-screen.
+ * Note: The web page must only contain one video tag in order for the HTML5 video ended event to work. This could be improved if needed (see Javascript code).
+ */
+
+open class VideoEnabledWebChromeClient(private var activityNonVideoView: View?, private var activityVideoView: ViewGroup?, private var loadingView: View?, private var webView: VideoEnabledWebView?) : WebChromeClient(), OnPreparedListener, OnCompletionListener, MediaPlayer.OnErrorListener {
+    interface ToggledFullscreenCallback {
```
Comment (MatyashDen) [inline app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt:40]: split by lines?

File: app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+     * @param callback A com.bogdan.codeforceswatcher.components.VideoEnabledWebChromeClient.ToggledFullscreenCallback callback
+     */
+    fun setOnToggledFullscreen(callback: ToggledFullscreenCallback?) {
+        toggledFullscreenCallback = callback
+    }
+
+    override fun onShowCustomView(view: View, callback: CustomViewCallback) {
+        if (view is FrameLayout) {
+            println("here frame")
+            // A video wants to be shown
```
Comment (MatyashDen) [inline app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt:68]: not needed?

File: app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                println("here videView")
+                // android.widget.VideoView (typically API level <11)
+
+                // Handle all the required events
+                focusedChild.setOnPreparedListener(this)
+                focusedChild.setOnCompletionListener(this)
+                focusedChild.setOnErrorListener(this)
+            } else {
+                println("here other")
+
```
Comment (MatyashDen) [inline app/src/main/java/com/bogdan/codeforceswatcher/components/VideoEnabledWebChromeClient.kt:91]: Not needed?

**Review by MatyashDen — APPROVED** [review]

**Pull request merged**